src/thread.rs

```diff
@@ -377,6 +377,10 @@ impl<'mir, 'tcx: 'mir> ThreadManager<'mir, 'tcx> {
     }
 
     /// Register the given `callback` to be called once the `call_time` passes.
+    ///
+    /// We made two design decisions when implementing timeout callbacks:
+    /// 1. The caller of the callback is responsible for changing the thread.
+    /// 2. The scheduler is the only place that can change the active thread.
```

Comment:
I was imagining something more client-focused, like
```suggestion
    /// The callback will be called with `thread` being the active thread,
    /// and the callback may not change the active thread.
```